### PR TITLE
Added commit to allow for midi values in files to exceed 127.

### PR DESCRIPTION
**Pull request by cjappl**
If "clip" is specified in midi file initializer, these values will be clipped to 127, otherwise, an error will be thrown. As referenced in issue #63

### CODE REVIEW

**Comment by olemb** [comment]
Looks good! I really appreciate that you've fixed little things you found along the way.

Some comments:

I see you've changed "Todo" to "TODO". Is that a more common way to write it? If we make the change we should probably update everywhere.

I forgot that you could do "not in". Neat and much more readable.

We could move the byte clipping into a `clip_data_bytes()` function. Not a big deal though.

At some point it would be nice to have more specific exceptions instead of `IOError` everywhere. `MidiParseError` or something would be more helpful (as a subclass of `IOError` or course). Same for ports (`PortNotFound` etc.). (It's on the roadmap somewhere.)

I think the "not all MIDI messages have running status" comment is out of date and can be removed.

The "`if (not msg.is_meta and raw[0] < 0xf0 and raw[0] == running_status_byte):`" is a bit long. I try to keep lines below 80 characters. Though if we keep it on one line the parenthesis can go. :)

I guess all that's missing now is the warning (and documentation).

Should I merge now or do you want to update the pull request?

File: mido/midifiles/midifiles.py [hunk 1]
```diff
@@ -117,19 +118,28 @@ def read_file_header(infile):
...
 
-    # Subtrac 1 for status byte.
+    # Subtract 1 for status byte.
     size = spec['length'] - 1 - len(peek_data)
     data_bytes = peek_data + read_bytes(infile, size)
 
-    for byte in data_bytes:
-        if byte > 127:
-            raise IOError('data byte must be in range 0..127')
+    adjusted_bytes = []
```
Comment (craffel) [inline mido/midifiles/midifiles.py]: You can replace line 131-138 with
```Python
if clip:
    data_bytes = [byte for byte in data_bytes if byte < 127 else 127]
```

File: mido/midifiles/midifiles.py [hunk 2]
```diff
@@ -334,8 +345,8 @@ def _load(self, infile):
                 if self.debug:
                     _dbg('Track {}:'.format(i))
 
-                self.tracks.append(read_track(infile, debug=self.debug))
-                # Todo: used to ignore EOFError. I hope things still work.
+                self.tracks.append(read_track(infile, debug=self.debug, clip=self.clip))
```
Comment (craffel) [inline mido/midifiles/midifiles.py]: @olemb do you enforce an 80-character width limit (PEP-8)?

File: mido/midifiles/midifiles.py (hunk 1 above)
Comment (cjappl) [inline mido/midifiles/midifiles.py]: Nice! That's much cleaner, thank you.

File: mido/midifiles/midifiles.py (hunk 2 above)
Comment (cjappl) [inline mido/midifiles/midifiles.py]: Fixed it! That's the one thing I have my linter turned off on, thanks for catching it :)

**Comment by cjappl** [comment]
@olemb I'll update this one right now!

> Looks good! I really appreciate that you've fixed little things you found along the way.

No problem! Maybe after this one I'll do a quick "clean up" pull request. I don't know if you know the tool [flake8](http://flake8.pycqa.org/en/latest/index.html#quickstart) but it's my preferred linter. It'll yell at you if anything is outside PEP8 and makes it really easy to clean up.

> I see you've changed "Todo" to "TODO". Is that a more common way to write it? If we make the change we should probably update everywhere.

Let me do this as a part of the "cleanup" review, just so I don't clutter this one. I know in my development environment, it boldly highlights TODO, but not Todo. Just makes them stand out better.

**Comment by olemb** [comment]
Thanks!

Let's go with TODO. I've had a look around and that seems to be more or less standard. We could also consider changing some of them to FIXME where that applies.

I'll have a look at flake8.

I should use linters more. I ran pyflakes it helped me find lots of little things like unused imports: 4d96942dcedba46d8ab4c35d03fcef215a2a0ed2

File: mido/midifiles/midifiles.py (hunk 2 above)
Comment (olemb) [inline mido/midifiles/midifiles.py]: @craffel Yes, I enforce an 80-character limit.

**Pull request merged**